src/services/submissionService.ts

```diff
@@ -16,20 +16,66 @@ export const getSubmissionById = async (id: string) : Promise<Submission | null>
 
         return convertedEntity
     } catch (error) {
-        throw new Error(`Failed to retrieve submission: ${error}.`)
+        throw new Error(`Error retrieving submission: ${error}`)
     }
 }
 
-function convertToDbSubmission(submission: import('../typeDefs/gqlTypes').Submission): 
-    import('../typeDefs/dbSchema').Submission {
+export async function getSubmissions(filters: dbSchema.SubmissionSearchFilters = {}) {
+    try {
+        const db = getFirestore()
+        let subRef: Query<DocumentData> = db.collection('submissions')
+
+        if (filters.googleMapsUrl) {
+            subRef = subRef.where('googleMapsUrl', '==', filters.googleMapsUrl)
+        }
+
+        if (filters.healthcareProfessionalName) {
+            subRef = subRef.where('healthcareProfessionalName', '==', filters.healthcareProfessionalName)
```

Comment:
In a future commit, we should have this do contains the part of the name instead of the whole name matching. We should also make it case-insensitive ideally, but I'm not sure firebase supports this. I'll have to look into it. 
If you're interested, feel free to t try and solve , otherwise I can figure it out later 🙂